internal/pkg/skuba/gangway/gangway.go

```diff
@@ -73,6 +77,28 @@ func CreateOrUpdateSessionKeyToSecret(client clientset.Interface, key []byte) er
 	return nil
 }
 
+// GetClientSecret returns the client secret from configmap
+func GetClientSecret(client clientset.Interface) (string, error) {
+	type config struct {
+		ClientSecret string `yaml:"clientSecret"`
+	}
+
+	cm, err := client.CoreV1().ConfigMaps(metav1.NamespaceSystem).Get(configmapName, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return "", nil
+		}
+		return "", err
+	}
+
+	data := cm.Data["gangway.yaml"]
```

Comment:
Suggest to check if the key exists, and if not just regard the cm does not exist.

```suggestion
	data, ok := cm.Data["gangway.yaml"]
	if !ok {
	    return "", nil
	}
```